code/app/src/main/java/com/example/sigma_blue/DebuggingActivity.java

```diff
@@ -0,0 +1,206 @@
+            this.itemListAdapter.addItem(
+                new Item(
+                        "ThinkPad", new Date(), "Nice UNIX book", "IBM",
+                        "T460", 300f
+                )
+            );
+
+            /* Updates the summation */
+            this.viewHolder.setSummaryView(itemListAdapter.sumValues());
+        });  // Launch add fragment.
+        viewHolder.searchButton.setOnClickListener(v -> {});    // Launch search fragment
+        viewHolder.sortFilterButton.setOnClickListener(v -> {
+             if (dbInit.checkExistence(new Account("Watrina",
+                    "lkj312"))) Toast.makeText(this,
+                     "Exists", Toast.LENGTH_SHORT).show();
+             else Toast.makeText(this, "Does not exist",
+                     Toast.LENGTH_SHORT).show();
+        });
+        viewHolder.optionsButton.setOnClickListener(v -> {
+
+            // FOR ItemDB TESTING PURPOSE
+
+            login(iDB, "testUser", "112233", this);
+            // after login, the
+
+            // build a mock item
+
+
+        });
+    }
+
+    @Override
+    public void login(ItemDB idb, String userName, String password, Context Activity) {
+        iDB.getDb().collection("SigmaBlue")
+                .document(userName)
+                .collection("AccountInfo")
+                .get()
+                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                        for (QueryDocumentSnapshot doc : task.getResult()) {
+                            if (doc.getId().compareTo("Password") == 0) {
+                                //check for the password
+                                if (doc.getString("Password").compareTo(password) == 0) {
+                                    iDB.setLoginUser(userName);
+                                    int duration = Toast.LENGTH_SHORT;
+                                    Toast.makeText(Activity, "Successful Login", duration).show();
+                                } else {
+                                    int duration = Toast.LENGTH_SHORT;
+                                    Toast.makeText(Activity, "Failed to Login", duration).show();
+                                }
+
+                            }
+                        }
+
+                        // TEST Save To DB after login; FOR TESTING ONLY
+                        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM");
+                        Date date = new Date();
+                        try {
+                            date = formatter.parse("2022-01");
+                        } catch (ParseException e) {
+                            Calendar calendar = Calendar.getInstance();
+                            calendar.setTime(date);
+                        }
+                        Item item = new Item("3090", date, "Testing", "Evga", "GA102-220-A1", (float)799.99);
```

Comment:
Creating an item must include the 'Comment' in the constructor

Reply:
I think that I will put a newInstance method into the Item class so that we could more easily add fields or change in the internals without affecting the constructors.

Reply:
Ah, fair. For the next pull request